src/users/helpers.ts

```diff
@@ -0,0 +1,60 @@
+import {Request, Response, NextFunction} from "express";
+import jwt from "jsonwebtoken";
+import {getConfig} from '../config';
+
+interface IUser {
+  isAuthenticated: boolean,
+  user: { role: string, token:string } | null
+}
+
+const { jwt: {secretKey}} = getConfig();
+
+export const verifyToken = (req: Request): IUser => {
+  const { authorization = '' } = req.headers
+  const token = authorization.slice(7)
+
+  if (!token) {
+    return {
+      isAuthenticated: false,
+      user: null
+    }
+  }
+
+  try {
+    const currentUser = jwt.verify(token, secretKey) as { role: string, token:string }
+    return {
+      isAuthenticated: true,
+      user: currentUser
+    }
+  } catch (err) {
+    return {
+      isAuthenticated: false,
+      user: null
+    }
+  }
+}
+
+export const checkUserIsAdmin = (req:Request, res:Response, next: NextFunction) => {
+  const currentUser = module.exports.verifyToken(req)
```

Comment:
Just access `verifyToken` directly instead of from the `exports` object, this just makes the code less verbose